addons/osfstorage/models.py

```diff
@@ -337,26 +337,13 @@ class OsfStorageFolder(OsfStorageFileNode, Folder):
 
     @property
     def is_checked_out(self):
-        try:
-            if self.checkout:
-                return True
-        except AttributeError:
-            return False
-        # TODO this should be one query
-        for child in self.children.all():
-            try:
-                if child.is_checked_out:
-                    return True
-            except AttributeError:
-                pass
-        return False
+        return self.children.filter(checkout__isnull=False).exists()
 
     @property
     def is_preprint_primary(self):
-        if self.node.preprint_file:
```

Comment:
I think the `self.node.preprint_file` check prevents the query from ever having to happen if it isn't a preprint. Removing it might slow down checks on nodes that are not preprints.